appinventor/components/src/com/google/appinventor/components/runtime/ChartDataModel.java

```diff
@@ -95,15 +97,47 @@ public void setElements(String elements) {
     }
 
     /**
-     * Imports data from a YailList which contains nested tuples
+     * Imports data from a List object.
+     * Valid tuple entries are imported, and the invalid entries are ignored.
      *
-     * @param list  YailList containing tuples
+     * @param list  List containing tuples
      */
-    public void importFromList(YailList list) {
-        // Iterate over all the tuples
-        for (int i = 0; i < list.size(); ++i) {
-            YailList tuple = (YailList)list.getObject(i);
-            addEntryFromTuple(tuple);
+    public void importFromList(List list) {
```

Comment:
Additional tests for ImportFromList:
- [ ] Generic List (with YailList entry)
- [ ] Generic List (with non-List entry)
- [ ] Generic List (with List entry)
- [ ] YailList (with non-List entry)
- [ ] YailList (with List entry)